roles/openshift_aws/tasks/uninstall_iam_cert.yml

```diff
@@ -0,0 +1,26 @@
+---
+- block:
+  - name: delete AWS IAM certificates
+    iam_cert23:
+      state: absent
+      name: "{{ openshift_aws_iam_cert_name }}"
+    register: elb_cert_chain
+    retries: 20
+    delay: 10
+    until: elb_cert_chain | succeeded
+    ignore_errors: yes
+
+  - debug:
+      var: elb_cert_chain
+      verbosity: 1
+
+  - name: check for iam cert error
+    fail:
+      msg: "Couldn't delete IAM cert {{ openshift_aws_iam_cert_name }}"
+    when: not elb_cert_chain | succeeded
+
+  when:
```

Comment:
Do we require a block for this code if its in its own file?

When using block statements, I'd prefer to set the `when` at the top.  Thoughts?

Reply:
Ok, that sounds fine.  Can you invert the block so when's land on top?  I think it reads cleaner.

Reply:
the problem with the parsing was with this format
```
- block:
  when:
  - conditionA
  - conditionB
  - name: taskA
  - name: taskB
```

Putting 'when' above 'block' gets the parsing straightened out

```
- when:
  - conditionA
  - conditionB
  block:
  - name: taskA
  - name: taskB
```